devlab/tests/cleanup.py

```diff
@@ -96,21 +96,34 @@ def clean_flavors(self):
     def clean_images(self):
         all_images = self.migration_utils.get_all_images_from_config()
         images_names = [image['name'] for image in all_images]
-
         for image in self.glanceclient.images.list():
             if image.name not in images_names:
                 continue
-            self.glanceclient.images.delete(self.get_image_id(image.name))
+            image_id = self.get_image_id(image.name)
+            self.glanceclient.images.delete(image_id)
+            self.delete_image_by_id_from_db(image_id)
             LOG.info('Image "%s" has been deleted', image.name)
 
+    def delete_image_by_id_from_db(self, image_id):
+        glance_mysql = self.mysql_connector('glance')
+        sql_delete_pattern = "DELETE FROM {table} WHERE image_id=:image_id"
+        tables_names = ["image_locations", "image_properties",
+                        "image_members"]
+        for table in tables_names:
+            sql_delete_pattern_table = sql_delete_pattern.format(table=table)
+            glance_mysql.execute(sql_delete_pattern_table, image_id=image_id)
+        sql_image_delete_by_id = "DELETE FROM images WHERE id=:image_id"
+        glance_mysql.execute(sql_image_delete_by_id, image_id=image_id)
+
     def clean_snapshots(self):
         snaps_names = [snapshot['image_name']
                        for snapshot in self.config.snapshots]
         for snapshot in self.glanceclient.images.list():
             if snapshot.name not in snaps_names:
                 continue
-            self.glanceclient.images.delete(
-                self.get_image_id(snapshot.name))
+            image_id = self.get_image_id(snapshot.name)
+            self.glanceclient.images.delete(image_id)
+            self.delete_image_by_id_from_db(image_id)
```

Comment:
you are using this 3 lines of code twice, i suggest move it into separate method like `delete_image_by_name`